### PR TITLE
Adding yaml + lazy execution

### PR DESCRIPTION
**Pull request by marcromeyn**

### CODE REVIEW

**Comment by ryxli** [comment]
Wondering when this could get merged?

File: src/nemo_run/config.py [hunk 1]
```diff
@@ -262,6 +267,17 @@ def __init__(
 
         super().__init__(fn_or_cls, *args, **new_kwargs)
 
+    @classmethod
```
Comment (hemildesai) [inline src/nemo_run/config.py:270]: Why is this required? `Config` object handles dicts.
```
In [1]: import fiddle as fdl

In [2]: fdl.Config(dict, a=5)
Out[2]: <Config[dict(a=5)]>

In [3]: a = fdl.Config(dict, a=5)

In [4]: fdl.build(a)
Out[4]: {'a': 5}
```

File: src/nemo_run/run/task.py [hunk 2]
```diff
@@ -61,6 +61,9 @@ def dryrun_fn(
 
 
 def direct_run_fn(task: Partial | Script, dryrun: bool = False):
+    if hasattr(task, "__is_lazy__"):
```
Comment (hemildesai) [inline src/nemo_run/run/task.py:64]: Shouldn't this be `is_lazy` to be consistent with the others?

File: src/nemo_run/test_utils.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (hemildesai) [inline src/nemo_run/test_utils.py:1]: This should be moved to the `test` folder.

Comment (marcromeyn) [reply src/nemo_run/test_utils.py:1]: Doesn't work for the CLI, that's why I moved it inside the package.

Comment (hemildesai) [reply src/nemo_run/test_utils.py:1]: This file will be included in installation if its inside `src`, which is not ideal. Why doesn't it work with the cli? Can you explicitly load the file in the tests?

**Review by hemildesai — APPROVED** [review]
